### PR TITLE
case-insensitive search

### LINKED ISSUE #400: Make string search case-insensitive by default
**Is your feature request related to a problem? Please describe.**
String search is case-sensitive in the API by default.

**Describe the solution you'd like**
Make search case-insensitive by default. Probably should use `COLLATE NOCASE` in the SQL query.

This is releated to #376  

**Describe alternatives you've considered**
NA

**Additional context**
https://www.hl7.org/fhir/search.html#string indicates search should be case-insensitive by default.

**Would you like to work on the issue?**
This would be a good starter issue.

### PR DESCRIPTION
**Pull request by maanuanubhav999**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straighforward changes).**

Fixes #400 

**Description**
added case insensitivity

**Alternative(s) considered**
N/A

**Type**
Choose one:  Feature

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledge the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md)
- [x] I have read [How to Contribute](https://github.com/google/android-fhir/blob/master/docs/contributing.md)
- [x] I have read the [Developer's guide](https://github.com/google/android-fhir/wiki/Developer's-Guide)
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate )
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally
- [x] I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

### CODE REVIEW

File: core/src/main/java/com/google/android/fhir/search/MoreSearch.kt [hunk 1]
```diff
@@ -70,7 +70,7 @@ fun Search.getQuery(): SearchQuery {
     SELECT a.serializedResource
     FROM ResourceEntity a
     $sortJoinStatement
-    WHERE a.resourceType = ?
+    WHERE a.resourceType collate nocase = ?
```
Comment (jingtang10) [inline core/src/main/java/com/google/android/fhir/search/MoreSearch.kt:73]: Are you sure this is not breaking the query?

Also please use all caps for keywords.

```suggestion
    WHERE a.resourceType = ? COLLATE NOCASE
```

https://www.designcise.com/web/tutorial/how-to-do-case-insensitive-comparisons-in-sqlite

**Review by jingtang10 — CHANGES_REQUESTED** [review]

**Comment by jingtang10** [comment]
Thanks for the contribution 👍  please see comment in review.

File: core/src/main/java/com/google/android/fhir/search/MoreSearch.kt (hunk 1 above)
Comment (maanuanubhav999) [reply core/src/main/java/com/google/android/fhir/search/MoreSearch.kt:73]: Sure @jingtang10  will follow that.

**Comment by maanuanubhav999** [comment]
@jingtang10 made the requested changes kindly review it again.

**Review by jingtang10 — APPROVED** [review]
Thanks!

**Review by jingtang10 — CHANGES_REQUESTED** [review]
unapprove.

the check should be on the string value in the string filter not on the resource type.

File: core/src/main/java/com/google/android/fhir/search/MoreSearch.kt [hunk 2]
```diff
@@ -70,7 +70,7 @@ fun Search.getQuery(): SearchQuery {
     SELECT a.serializedResource
     FROM ResourceEntity a
     $sortJoinStatement
-    WHERE a.resourceType = ?
+    WHERE a.resourceType = ? COLLATE NOCASE
```
Comment (jingtang10) [inline core/src/main/java/com/google/android/fhir/search/MoreSearch.kt:73]: this needs to be on line 88

**Review by jingtang10 — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/google/android/fhir/search/MoreSearch.kt (hunk 2 above)
Comment (maanuanubhav999) [reply core/src/main/java/com/google/android/fhir/search/MoreSearch.kt]: done

Comment (maanuanubhav999) [reply core/src/main/java/com/google/android/fhir/search/MoreSearch.kt]: thanks!

**Review by jingtang10 — APPROVED** [review]

**Pull request merged**